### PR TITLE
Fixing lack of firing flush timer in -initWithToken:launchOptions:andFlushInterval: method.

### PR DESCRIPTION
**Pull request by Grubas7**
Fix for issue https://github.com/mixpanel/mixpanel-iphone/issues/241

### CODE REVIEW

**Comment by samgreen** [comment]
Hi @Grubas7, thank you so much for the pull request. We really appreciate when the community wants to give back. 

Unfortunately this change will alter the public facing API in a non intuitive way, and could possibly break code that is depending on existing behavior. We are currently discussing other options internally such as providing an additional constructor that will immediately start the flush timer.

**Comment by Grubas7** [comment]
IMO right now work of SDK is non intuitive. I think @originell and @uooq would agree with me ;)
Why? When you create Mixpanel instance by `-initWithToken:andFlushInterval:` method there won't be any flush after time you could expect. And there's no public method to manually start/stop flushing timer. Only possible way to force SDK to periodically flush is set property `-flushInterval:` (might be even default value) after init.

In other words:
`-initWithToken:andFlushInterval:` doesn't start flushing timer
`-flushInterval:` starts flushing timer
Same parameter name, but different reaction.

**Comment by alex-hofsteede** [comment]
@Grubas7 yeah we agree that the current behavior is not very intuitive, but we also need to evaluate whether changing it would break Mixpanel implementations for customers that are currently relying on the existing behavior, which has existed for some time. Pushing a behavior change like this is therefore not as trivial as it might seem. 

If we were to implement this change, we would also need to consider situations where e.g. a Mixpanel instance is being initialized during a background task, and whether we should still automatically start the flush timer in that case.

**Comment by Grubas7** [comment]
@samgreen @alex-hofsteede As long as `-flushInterval:` property starts timer there's no problem for me ;) But maybe putting this issue to 'know issues' would be helpful for other devs.

**Comment by samgreen** [comment]
I'm going to go ahead and clearly document the behavior for now. We can consider changing the internal behavior in the next major release.

**Comment by originell** [comment]
Thanks for doing that! :)
